src/puppetlabs/puppetdb/anonymizer.clj

```diff
@@ -31,11 +32,10 @@
    (contains? event "old_value")
    (contains? event "resource_type")))
 
-(defn report?
+(pls/defn-validated report? :- report/report-schema
```

Comment:
So @wkalt had also mentioned we should do this and I changed it but I think this would require us to use `string` keys instead of `keyword` keys everywhere inside the anonymizer code which is a pretty significant change and would be inconsistent with catalogs for instance.